### PR TITLE
Add KMS support to bigquery tables.

### PR DESCRIPTION
**Pull request by ndmckinley**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/2014.

### CODE REVIEW

File: google/resource_bigquery_table.go [hunk 1]
```diff
@@ -347,6 +367,13 @@ func resourceBigQueryTableRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if res.EncryptionConfiguration != nil {
+		err := d.Set("encryption_configuration", flattenEncryptionConfiguration(res.EncryptionConfiguration))
```
Comment (danawillow) [inline google/resource_bigquery_table.go]: nit: `if err := stuff; err != nil {`

File: google/resource_bigquery_table_test.go [hunk 2]
```diff
@@ -236,6 +267,96 @@ EOH
...
+	org_id          = "%s"
+	billing_account = "%s"
+}
+
+resource "google_project_services" "acceptance" {
+	project = "${google_project.acceptance.project_id}"
+
+	services = [
+	  "cloudkms.googleapis.com",
+		"bigquery-json.googleapis.com",
```
Comment (danawillow) [inline google/resource_bigquery_table_test.go]: nit: tabs

File: google/resource_bigquery_table_test.go [hunk 3]
```diff
@@ -236,6 +267,96 @@ EOH
...
+resource "google_bigquery_dataset" "test" {
+	project = "${google_project.acceptance.project_id}"
+  dataset_id = "%s"
+	depends_on = ["google_project_services.acceptance"]
+}
+
+resource "google_kms_crypto_key_iam_member" "allow" {
+	crypto_key_id = "${google_kms_crypto_key.crypto_key.self_link}"
+	role = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+	member = "serviceAccount:bq-47465025999@bigquery-encryption.iam.gserviceaccount.com"
```
Comment (danawillow) [inline google/resource_bigquery_table_test.go]: this number looks like a project number, should it be hardcoded in this way?

**Comment by ndmckinley** [comment]
Sorry, this wasn't ready for review yet!  I'll get it ready and ping you.

File: google/resource_bigquery_table_test.go (hunk 2 above)
Comment (ndmckinley) [reply google/resource_bigquery_table_test.go]: I'm gonna disagree and say nit: spaces.  :stuck_out_tongue: 

Fixed it, though!

File: google/resource_bigquery_table.go (hunk 1 above)
Comment (ndmckinley) [reply google/resource_bigquery_table.go]: Done!  I didn't know we preferred that style.

**Comment by rileykarson** [comment]
This PR looks outdated after the move to MM. Do you mind closing this out + making it there?

**Comment by danawillow** [comment]
Opening this back up since the work has been done, but just needs to be moved over to MM (and possibly cleaned up a bit)

**Comment by MartinNowak** [comment]
It's a pitty that this has been lingering here for 5 months. Could @rileykarson or @danawillow shine some light on what exactly needs to be done here.

**Comment by danawillow** [comment]
@ndmckinley since this is yours, mind taking care of moving it over?

**Comment by danawillow** [comment]
Closing this since it's been upstreamed.

**Pull request closed without merging**